src/main/java/org/opensearch/neuralsearch/processor/chunker/ChunkerFactory.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.chunker;
+
+import java.util.Set;
+
+import org.opensearch.index.analysis.AnalysisRegistry;
+
+/**
+ * A factory to create different chunking algorithm classes and return all supported chunking algorithms.
+ */
+public class ChunkerFactory {
+
+    public static final String FIXED_LENGTH_ALGORITHM = "fix_length";
+    public static final String DELIMITER_ALGORITHM = "delimiter";
+
+    public static FieldChunker create(String type, AnalysisRegistry analysisRegistry) {
```

Comment:
Suggest change the class name to without `Field` since we can only chunk a field value, this name can cause confusion, e.g. we have other non field type chunkers.

Reply:
Agreed. Will change this.

Reply:
Already changed